src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,136 @@
+import { SSCIconNames } from '../../theme/icons/icons.enums';
+
+const widthVariants = {
+  [ModalWidthSizes.xs]: 320,
+  [ModalWidthSizes.sm]: 430,
+  [ModalWidthSizes.md]: 600,
+  [ModalWidthSizes.lg]: 900,
+};
+
+const Overlay = styled.div`
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  top: 0;
+  left: 0;
+  background: rgba(0, 0, 0, 0.6);
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  z-index: ${getDepth('modal')};
+`;
+
+const BaseModal = styled.div<{ $width: number }>`
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  max-height: 66vh;
+  width: ${({ $width }) => pxToRem($width)};
+  border-radius: ${getBorderRadius};
+  background-color: ${getColor('graphite5H')};
+`;
+
+const Title = styled(H3)`
+  margin: 0;
+`;
+
+// TODO Extract close button and unify it across DS
+const CloseButton = styled.button`
+  position: absolute;
+  top: ${pxToRem(8)};
+  right: ${pxToRem(8)};
+  width: ${pxToRem(16)};
+  height: ${pxToRem(16)};
+  ${({ theme }) => createPadding({ paddingSize: SpaceSizes.md, theme })};
+  font-size: ${getFontSize('lg')};
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  flex-shrink: 0;
+  border: none;
+  cursor: pointer;
+  color: ${getColor('graphite3B')};
+`;
+
+const Content = styled(Padbox)`
+  padding-top: 0;
+  overflow-y: auto;
+`;
+
+const Footer = styled(Padbox)`
+  border-top: 1px solid ${getColor('graphiteHB')};
+`;
+
+const Modal = forwardRef<HTMLDivElement, ModalProps>(
+  ({ title, onClose, footer, width = ModalWidthSizes.md, children }, ref) => {
+    const { portalsContainerId } = useContext(DSContext);
+    const { Portal } = usePortal({
+      containerId: portalsContainerId,
+      internalShowHide: false,
+    });
+
+    const modalRef = useOuterClick<HTMLDivElement>(onClose);
+
+    useLockBodyScroll();
+
+    return (
+      <Portal>
+        <Overlay>
+          <BaseModal
+            ref={mergeRefs<HTMLDivElement>(modalRef, ref)}
+            $width={widthVariants[width]}
+          >
+            <CloseButton aria-label="Close" onClick={onClose}>
+              <Icon name={SSCIconNames.times} />
+            </CloseButton>
+            <Padbox paddingSize={SpaceSizes.lg} paddingType="squish">
+              {isNotUndefined(title) && <Title>{title}</Title>}
+            </Padbox>
+            <Content paddingSize={SpaceSizes.lg} paddingType="squish">
+              {children}
+            </Content>
+            {isNotUndefined(footer) && (
+              <Footer paddingSize={SpaceSizes.lg} paddingType="squish">
+                {footer}
+              </Footer>
+            )}
+          </BaseModal>
+        </Overlay>
+      </Portal>
+    );
+  },
+);
+
+export default Modal;
+
+Modal.propTypes = {
+  children: PropTypes.node.isRequired,
+  onClose: PropTypes.func.isRequired,
+  title: PropTypes.string,
+  footer: PropTypes.node,
+  width: PropTypes.oneOf(Object.values(ModalWidthSizes)),
```

Comment:
I would probably change this prop to `size` since you are not passing the width directly.

Reply:
Yep. `size` it is